torchtune/modules/transforms/tokenizers/_hf_tokenizer.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import json
+from typing import Any, Dict, List, Optional
+
+from tokenizers import Tokenizer
+from torchtune.modules.transforms.tokenizers._utils import BaseTokenizer
+
+
+class HFTokenizer(BaseTokenizer):
+    """
+    A wrapper around Hugging Face tokenizers. This class can be used to load from a
+    Hugging Face tokenizer.json file into a torchtune BaseTokenizer.
+
+    This class will load the tokenizer.json file from tokenizer_json_path. It will
+    attempt to infer BOS and EOS token IDs from config.json if possible, and if not
+    will fallback to inferring them from generation_config.json.
+
+    Args:
+        tokenizer_json_path (str): Path to tokenizer.json file
+        config_path (Optional[str]): Path to tokenizer_config.json file. Default: None
+        generation_config_path (Optional[str]): Path to generation_config.json file.
+            Default: None
+
+    Raises:
+        ValueError: If neither config_path or generation_config_path are specified.
+    """
+
+    def __init__(
+        self,
+        tokenizer_json_path: str,
+        config_path: Optional[str] = None,
+        generation_config_path: Optional[str] = None,
+    ):
+        self.hf_tokenizer = Tokenizer.from_file(tokenizer_json_path)
+        if not (config_path or generation_config_path):
+            raise ValueError(
+                "At least one of config_path or generation_config_path must be specified."
+            )
+        if config_path:
+            with open(config_path, "rb") as f:
+                self.config = json.load(f)
+        else:
+            self.config = None
+        if generation_config_path:
+            with open(generation_config_path, "rb") as f:
+                self.generation_config = json.load(f)
+        else:
+            self.generation_config = None
+        self._infer_bos_eos_tokens()
```

Comment:
Should we give them an option to explicitly set the BOS and EOS if it can't be inferred from the file? Or are they just screwed?

Reply:
I am hedging a bit here for future work -- eventually I would like to have a `ModelTokenizer` wrapping this. In that case I think we should require at least `tokenizer_config.json` and explicitly giving EOS and BOS will probably not be enough